src/metabase/integrations/slack.clj

```diff
@@ -154,7 +157,7 @@
   default only fetches channels."
   [& {:as query-parameters}]
   (let [params (merge {:exclude_archived true, :types "public_channel"} query-parameters)]
-    (paged-list-request "conversations.list" :channels params)))
+    (vec (paged-list-request "conversations.list" :channels params))))
```

Comment:
As mentioned above, there are a few reasons why I think forcing this to be fully realized _here_ is a bad idea.

1. This forces us to have the entire API response in memory, even if we only need one key from each object in the results (as is the actual case). This means this ends up taking a huge amount of memory for big API responses
2. If we only wanted one or two items this still ends up fetching potentially millions of items. `(take 1 infinite-lazy-sequence)` will just return the first item and will not force the rest of the `infinite-lazy-sequence` to be realized. `(take 1 (conversations-list))` will never fetch more than one page if `(conversations-list)` is lazy; if it's not we're always forced to fetch every single page regardless of how many we actually _want_.